File: problem-3/main.py
import math

def largest_prime_factor(x):
    large_prime = 1
    while x % 2 == 0:
        large_prime = 2
        x //= 2  
    

    for i in range(3, int(math.sqrt(x)) + 1, 2):
        while x % i == 0:
            large_prime = i
            x //= i
    if x > 2:
        large_prime = x
    
    return large_prime

x = 600851475143
print(largest_prime_factor(x))

File: problem-3/main.go
package main

import (
	"fmt"
	"math"
)

func largestPrimeFactor(x int64) int64 {
	var largePrime int64 = 1
	for x%2 == 0 {
		largePrime = 2
		x /= 2
	}
	for i := int64(3); i <= int64(math.Sqrt(float64(x))); i += 2 {
		for x%i == 0 {
			largePrime = i
			x /= i
		}
	}
	if x > 2 {
		largePrime = x
	}

	return largePrime
}

func main() {
	x := int64(600851475143)
	fmt.Println(largestPrimeFactor(x))
}
